submissions/kt-std/html-css-popup/css/style.css

```diff
@@ -0,0 +1,370 @@
+	cursor: pointer;
+}
+
+a:focus,
+.popup,
+.more-btn,
+.checkbox__image:hover,
+.checkbox__input:focus + .checkbox{
+	border-radius: 5px;
+}
+
+a:focus,
+.checkbox__input:focus + .checkbox{
+	border: 1px dashed #85c295;
+}
+
+
+
+.checkbox__input,
+.checkbox__input_bell{
+	-moz-appearance:none;
+	-webkit-appearance: none;
+	appearance:none;
+	width: 1px;
+	position: absolute;
+	left: -1px;
+	height: 1px;
+}
+
+.checkbox__image{
+	width: 25px;
+	height: 25px;
+}
+.checkbox__image:hover{
+	background-color: rgba(100, 100, 100, .1);
+}
+
+.bell{
+	padding: 20px;
+	filter: saturate(0);
+}
+.bell:hover{
+	background-color: white;
+}
+
+.bell_input:checked + .checkbox>.bell{
+	filter: saturate(1);
+	animation: ring 1.5s ease;
+}
+
+
+@keyframes ring {
+    0% {
+        transform: rotate(35deg);
+    }
+    12.5% {
+        transform: rotate(-30deg);
+    }
+    25% {
+        transform: rotate(25deg);
+    }
+    37.5% {
+        transform: rotate(-20deg);
+    }
+    50% {
+        transform: rotate(15deg);
+    }
+    62.5% {
+        transform: rotate(-10deg);
+    }
+    75% {
+        transform: rotate(5deg);
+    }
+    100% {
+        transform: rotate(0deg);
+    }
+}
+
+.item__link,
+.button,
+.search-container,
+.checkbox{
+	height: 100%;
+}
+
+.item__link{
+	width: 90px;
+	padding: 10px 0;
+}
+
+.item__icon {
+	min-width: 45px;
+	margin-bottom: 10px;
+	height: 45px;
+	background-size: contain;
+	background: center no-repeat;
+}
+
+.no-padding{
+	padding: 0;
+}
+
+.popup,
+.popup__more{
+	display: none;
+	justify-content: space-around;
+	border-top: 1px solid #e0e0e0;
+	padding-top: 20px;
+}
+
+.popup {
+	overflow-y: auto;
+	flex-wrap: wrap;
+	position: absolute;
+	z-index: 2;
+	bottom: 0;
+	right: 0;
+	transform: translate(0, 100%);	
+	max-height: 420px;
+	width: 320px;
+	border: 1px solid #e0e0e0;
+	background-color: white;
+}
+
+.popup-btn{
+	position: relative;
+}
+
+.popup__item {
+	margin-bottom: 20px;	
+}
+
+.logo{
+	min-width: 200px;
+	margin: -80px 0 20px 0;
+	max-width: 500px;
+	height: auto;
+}
+
+.button:hover .button__image,
+.search-field__input:hover ~ 
+.search-field__button>.button__image{
+	fill: rgba(76, 96, 137,.6);
+}
+
+.search-field__input:focus ~ 
+.search-field__button>.button__image{
+	fill: rgba(77, 182, 172, .7);
+}
+
+.search-field {
+	position: relative;
+	max-width: 500px;
+}
+
+.search-field__input {
+	height: 50px;	
+	font-family: Lato, sans-serif;
+	border-radius: 25px;
+	border: none;
+	text-transform: capitalize;
+	box-shadow: 5px 5px 25px rgba(0,0,0,.2); 
+	padding: 25px;
+	transition: box-shadow 0.5s ease-in; 
+}	
+
+.search-field__input:hover{
+	box-shadow: 5px 5px 20px rgba(76, 96, 137,.6);
+} 
+
+.search-field__input:focus{
+	box-shadow: 5px 5px 20px rgba(77, 182, 172, .7);
+}
+
+.search-field__button {
+	background-color: white;
+	border: none;
+	outline: none;
+	border-radius: 50%;
+}
+
+.button{
+	position: absolute;
+	width: 60px;
+	z-index: 1;
+	right: 0;
+}
+
+button::-moz-focus-inner {
```

Comment:
And please don't use html tags in css.
Read this article for more understanding: https://frontstuff.io/you-need-to-stop-targeting-tags-in-css

Reply:
I was thinking of this case while writing the code. I tore between the usage of tag and the class. I've already updated the code with class, but could you please explain what solution would be better in that case? The case actually is: I decided to use the tag instead of the class because this rule fixes the default outline of a button in Firefox (shown on the screenshot below) so that the style should be applied to all button tags. 
![fix](https://user-images.githubusercontent.com/74820334/100541502-03a5cc00-324d-11eb-9ff6-fec397d3d6e5.png)
I guess, it matches the idea from the article you sent:

> Now following that logic, there’s one case when styling on tags can actually make sense: when absolutely all tags of a certain type need to be styled the same way (or when there’s only one of them in the DOM, like HTML or body).  

Therefore, _to tag_ or _not to_ _tag_?

Reply:
Imagine that not only you will support or upgrade this application. And this app will have 10000 lines of code (in additional files too). And someone knows nothing about the styling of tags and will try to add pictures, links, (whatever) with their own styles and will not understand why some of them do not apply or apply not as was desirable
It is a just good practice

Reply:
Thank you for clarifying! :)